execution_workflows/APA-Scan/workflow/Snakefile

```diff
@@ -0,0 +1,357 @@
+# BY APA-SCAN. IE FOR THE TEST DATA, THIS MAKES SO SRSF3 is CONDITION1 AND
+# THE CONTROL IS CONDITION 2
+
+################################################################################
+
+
+
+# Parameters
+THREADS = 8
+
+
+# Input file directories
+CONDITION1 = samples["sample_type"].unique()[0]
+CONDITION2 = samples["sample_type"].unique()[1]
+
+
+# Output filename shorthands
+APA_SCAN_OUTPUT = "APA_Scan_" + CONDITION1 + "_Vs_" + CONDITION2 + ".xlsx" # Raw output of APA-Scan.py
+APA_SCAN_OUTPREFIX = "AA_APA-Scan" # Prefix for final APA-Scan output
+
+# These NEED to be in the same directory you run APA-Scan.py from.
+ANNOTATION = "ncbiRefSeq.txt"
+GENOMEFILE = "mm10.fa"
+
+
+# Print to compare conditions and check correct way round
+print(">>>>> COMPARING: ")
+print("   >> " + CONDITION1)
+print("   >> " + CONDITION2)
+print("   >> APA-Scan RAW OUTPUT IS :- " + config["out_dir"] + "/" + APA_SCAN_OUTPUT)
+
+
+
+
+
+# Final Target Rule ------------------------------------------------------------
+localrules: finish
+
+# Final output .xlsx file, that's been renamed with rename_output.
+
+rule finish:
+    """Rule that specifies the final output.
+    """
+    input:
+        os.path.join(config["out_dir"], APA_SCAN_OUTPREFIX + "_01.bed"),
+        os.path.join(config["out_dir"], APA_SCAN_OUTPREFIX + "_03.tsv")
+
+
+
+
+
+#-------------------------------------------------------------------------------
+# Preprocessing: obtain suitable input formats
+
+
+#### NOTES #####################################################################
+
+# What's going on here is that APA-Scan is set up so it reads a specific file,
+# "configuration.ini" to get the information it needs to run, such as where
+# the data, annotation and genome files are and how to output them.
+# What this function below does is there's a "blank" configuration file present
+# which has fields that correspond to different directories and files etc. These
+# are in the form "dataofinterest_dir", for example input_dir1, input_dir2,
+# annotation_dir and genome_dir. Using sed, we replace these tags with our
+# dirs/files of interest, which are set at the top. Output is just set to the
+# curren working directory as APA-Scan again is really painful and picky so
+# best not to tempt fate and the bioinformatics demons.
+
+# It's s obviously possible to just
+# define the configuration.ini file before running the whole pipeline, but this
+# wont work for multiple runs with different samples and studies. This is my
+# attempt to make it a bit generalisable (touch wood). I'm hoping whatever levels
+# you want to compare, you could parse that in some way from the previous step?
+
+################################################################################
+
+
+
+rule generate_configurationfile:
+    """A rule that does some preprocessing.
+    This example copies input to output and writes stdout and stderr to log.
```

Comment:
change this rule description. Notes from the paragraph above could go here (in a shortened form maybe)